### PR TITLE
PLT-4535/PLT-4503 Fix inactive users in searches and add option functionality to DB user search

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Does two things:
1. Add the infrastructure for passing options to the user DB search function
2. Adds an option for allowing inactive users to show up in the search results

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4535
https://mattermost.atlassian.net/browse/PLT-4503

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -2592,17 +2592,29 @@ func sanitizeProfile(c *Context, user *model.User) *model.User {
 }
 
 func searchUsers(c *Context, w http.ResponseWriter, r *http.Request) {
-	props := model.MapFromJson(r.Body)
+	props := model.StringInterfaceFromJson(r.Body)
```
Comment (crspeller) [inline api/user.go]: Can we unmarshal this into a struct instead? (could even be anonymous if you want)

Comment (jwilander) [inline api/user.go]: Ah, yeah that's a much better way of doing it

**Comment by jwilander** [comment]
Alright, I could probably streamline the whole thing a bit with more work but I think this more than good enough for now. Please re-review at your leisure @crspeller

File: api/user.go [hunk 2]
```diff
@@ -2592,33 +2593,45 @@ func sanitizeProfile(c *Context, user *model.User) *model.User {
 }
 
 func searchUsers(c *Context, w http.ResponseWriter, r *http.Request) {
-	props := model.MapFromJson(r.Body)
+	var props struct {
```
Comment (coreyhulen) [inline api/user.go]: Why isn't this is the model dir?

File: store/sql_user_store.go [hunk 3]
```diff
@@ -1097,8 +1099,9 @@ func (us SqlUserStore) Search(teamId string, term string, searchType string) Sto
 			FROM
 				Users
 			WHERE
-				DeleteAt = 0
+				Id != ''
```
Comment (coreyhulen) [inline store/sql_user_store.go]: how is it possible to have a blank userId?

**Comment by jwilander** [comment]
Alright @coreyhulen, I moved the anonymous struct over to model dir and used it instead of passing map[string]interface{} all over

File: store/sql_user_store.go (hunk 3 above)
Comment (coreyhulen) [inline store/sql_user_store.go]: 0/5 maybe we should add a note to the code, otherwise during a refactor I would probably delete that :)

Comment (jwilander) [inline store/sql_user_store.go]: Yeah, I'll add a quick comment

**Pull request merged**